**1. Réalisez une veille sur l’outil Jupyter Notebook** et installez le sur votre machine (soit en passant par Anaconda, soit directement sur VSCode).<br>
Familiarisez vous avec les cellules de code ainsi que les cellules de Markdown.<br><br>
**2. Créez et ouvrez un notebook Jupyter.** Nommez le “Netflix Data Analysis”.<br><br>
**3. Chargez le dataset à l’aide de Pandas** sous le format d’un DataFrame<br>
**Importe** la librairie *pandas*<br>
**Importe** les données depuis le fichier csv avec la fonction *read_*<br>
**Initialise** une variable *df* qui contient l'objet dataframe<br>

In [5]:
import pandas as pd
df=pd.read_csv('netflix_titles.csv')

**4. Affiche les 5 premières entrée du dataframe** via la variable df.

In [6]:
df.head()

show_id     type                  title         director  \
0      s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1      s2  TV Show          Blood & Water              NaN   
2      s3  TV Show              Ganglands  Julien Leclercq   
3      s4  TV Show  Jailbirds New Orleans              NaN   
4      s5  TV Show           Kota Factory              NaN   

                                                cast        country  \
0                                                NaN  United States   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                                NaN            NaN   
4  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   

           date_added  release_year rating   duration  \
0  September 25, 2021          2020  PG-13     90 min   
1  September 24, 2021          2021  TV-MA  2 Seasons   
2  September 24, 2021          2021  TV-MA   1 Season   
3  September 24, 2021          2021  TV-MA   1 Season   
4  September 24, 2021          2021  TV-MA  2 Seasons   

                                           listed_in  \
0                                      Documentaries   
1    International TV Shows, TV Dramas, TV Mysteries   
2  Crime TV Shows, International TV Shows, TV Act...   
3                             Docuseries, Reality TV   
4  International TV Shows, Romantic TV Shows, TV ...   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...  
2  To protect his family from a powerful drug lor...  
3  Feuds, flirtations and toilet talk go down amo...  
4  In a city of coaching centers known to train I...

**4 bis. Affiche les 5 dernières entrée du dataframe** via la variable df.

In [7]:
df.tail()

show_id     type        title         director  \
8802   s8803    Movie       Zodiac    David Fincher   
8803   s8804  TV Show  Zombie Dumb              NaN   
8804   s8805    Movie   Zombieland  Ruben Fleischer   
8805   s8806    Movie         Zoom     Peter Hewitt   
8806   s8807    Movie       Zubaan      Mozez Singh   

                                                   cast        country  \
8802  Mark Ruffalo, Jake Gyllenhaal, Robert Downey J...  United States   
8803                                                NaN            NaN   
8804  Jesse Eisenberg, Woody Harrelson, Emma Stone, ...  United States   
8805  Tim Allen, Courteney Cox, Chevy Chase, Kate Ma...  United States   
8806  Vicky Kaushal, Sarah-Jane Dias, Raaghav Chanan...          India   

             date_added  release_year rating   duration  \
8802  November 20, 2019          2007      R    158 min   
8803       July 1, 2019          2018  TV-Y7  2 Seasons   
8804   November 1, 2019          2009      R     88 min   
8805   January 11, 2020          2006     PG     88 min   
8806      March 2, 2019          2015  TV-14    111 min   

                                           listed_in  \
8802                  Cult Movies, Dramas, Thrillers   
8803          Kids' TV, Korean TV Shows, TV Comedies   
8804                         Comedies, Horror Movies   
8805              Children & Family Movies, Comedies   
8806  Dramas, International Movies, Music & Musicals   

                                            description  
8802  A political cartoonist, a crime reporter and a...  
8803  While living alone in a spooky town, a young g...  
8804  Looking to survive in a world taken over by zo...  
8805  Dragged from civilian life, a former superhero...  
8806  A scrappy but poor boy worms his way into a ty...

**5. Affichez les informations du DataFrame**<br> notamment le type d'index et
les colonnes, les valeurs non nulles et l'utilisation de la mémoire.

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8807 entries, 0 to 8806
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       8807 non-null   object
 1   type          8807 non-null   object
 2   title         8807 non-null   object
 3   director      6173 non-null   object
 4   cast          7982 non-null   object
 5   country       7976 non-null   object
 6   date_added    8797 non-null   object
 7   release_year  8807 non-null   int64 
 8   rating        8803 non-null   object
 9   duration      8804 non-null   object
 10  listed_in     8807 non-null   object
 11  description   8807 non-null   object
dtypes: int64(1), object(11)
memory usage: 825.8+ KB


**6. Affiche la dimensionnalité du DataFrame.**<br>
*Variables* : Les variables sont les colonnes du DataFrame. Il y en a 12.<br>
*Observation* : Les observations sont les lignes du DataFrame. Il y en a 8807.

In [9]:
dimensions = df.shape
rows, columns = dimensions
print(rows, columns)

8807 12


**7. Affichez les colonnes du DataFrame.**


In [11]:
print(df.columns)

Index(['show_id', 'type', 'title', 'director', 'cast', 'country', 'date_added',
       'release_year', 'rating', 'duration', 'listed_in', 'description'],
      dtype='object')


8. **Affichez le type des différentes colonnes du DataFrame.**

In [1]:
# méthode dtypes : affiche les types des différentes colonnes d'un DataFrame pandas
print(df.dtypes)

NameError: name 'df' is not defined

**Avez-vous des données quantitatives (numériques) ?**
<br>Si oui, lesquelles ?

In [22]:
# Sélection les colonnes numériques
digital_columns = df.select_dtypes(include=['int64', 'float64'])

if len(digital_columns.columns) > 0:
    print("\nColonnes numériques :\n")
    print(digital_columns)
else:
    print("\nAucune colonne numérique dans le DataFrame.\n")


Colonnes numériques :

      release_year
0             2020
1             2021
2             2021
3             2021
4             2021
...            ...
8802          2007
8803          2018
8804          2009
8805          2006
8806          2015

[8807 rows x 1 columns]


**Avez-vous des données qualitatives (catégorielles)**<br>Si oui, lesquelles ?

In [25]:
# Sélection les colonnes catégorielles
categorical_columns = df.select_dtypes(include=['object', 'bool'])


if len(categorical_columns.columns) > 0:
    print("Colonnes catégorielles :")
    print(categorical_columns)
else:
    print("Aucune colonne catégorielle dans le DataFrame.")

Colonnes catégorielles :
     show_id     type                  title         director  \
0         s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1         s2  TV Show          Blood & Water              NaN   
2         s3  TV Show              Ganglands  Julien Leclercq   
3         s4  TV Show  Jailbirds New Orleans              NaN   
4         s5  TV Show           Kota Factory              NaN   
...      ...      ...                    ...              ...   
8802   s8803    Movie                 Zodiac    David Fincher   
8803   s8804  TV Show            Zombie Dumb              NaN   
8804   s8805    Movie             Zombieland  Ruben Fleischer   
8805   s8806    Movie                   Zoom     Peter Hewitt   
8806   s8807    Movie                 Zubaan      Mozez Singh   

                                                   cast        country  \
0                                                   NaN  United States   
1     Ama Qamata, Khosi Ngema, Gail Mabalane, 

**9. Y a-t-il des données manquantes ?**<br> Identifiez la proportion en
pourcentage.

In [7]:

missing_data = df.isna()

# Calculer la proportion en pourcentage
percentage_missing = missing_data.mean() * 100

# Afficher les proportions en pourcentage
print("Proportion de données manquantes en pourcentage :")
print(percentage_missing)

Proportion de données manquantes en pourcentage :
show_id          0.000000
type             0.000000
title            0.000000
director        29.908028
cast             9.367549
country          9.435676
date_added       0.113546
release_year     0.000000
rating           0.045418
duration         0.034064
listed_in        0.000000
description      0.000000
dtype: float64
